getting_started_modules.md

```diff
@@ -0,0 +1,171 @@
+1. Write the code that uses the module, and run it manually inside the virtual machine 
+2. Write ansible code that'll run it automatically when you start the box.
+
+Below is an example on how to use a module:
+
+ ```hcl-terraform
+module "whatever" {
+  source      = "github.com/fredrikhgrelland/terraform-nomad-minio.git?ref=0.0.3"
+}
+```
+
+This will fetch the module at the given source. In the case above it is a MinIO module, version 0.0.3. 
+
+
+##### a. How To Use The Module and Run It Manually
+Create a file called `main.tf` under the `example/` directory, and add the code above, change the module's name, and source. The source is `../` in this case because that is where our module files are, relative to the `main.tf` you are writing this in. 
+
+Let's log onto our virtual machine and try and run it! Run `vagrant ssh` if needed, and navigate to your `/vagrant/example/` folder. Next run `terraform init` to initialize a terraform-workspace. When that is done you can try `terraform plan`, which will read your terraform code and attempt to make a runtime plan. When doing so you will get the error "Error: Missing required argument The argument "address" is required, but was not set.". This is because we are using a resource that deploys a nomad job, but nowhere in our terraform files have we defined _ what_ Nomad to use. At the moment no Nomad is known. This is where [providers]() come into the picture. They are providers for the resources we are using, and in our case we need to define a [Nomad provider](). We could do this in either of our `main.tf` files, but if we do it in our module's `main.tf` it will be very difficult for anyone to use our module, because the Nomad's address is predefined. Instead we should include what nomad to use in our example´s `main.tf`, which is simply an example of how to use the module, meaning anyone else wanting to use the module could supply their own nomad when using the module. To supply a provider add the lines below to your `example/main.tf` file:
+
+```hcl-terraform
+provider "nomad" {
+  address = "http://127.0.0.1:4646"
+}
+```
+
+we have now told terraform what Nomad we want to use. Try running your terraform code with `terraform init` (we need to load the nomad provider), `terraform plan` (this time it should succeed), then lastly `terraform apply`, which will execute the plan. Go to `localhost:4646` to check if the nomad-job has started running, if it has, congratulations, you have made your first working terraform module!
+
+
+
+##### b. Using Ansible To Run the Code in the Previous Step On Startup
+Way earlier I mentioned that ALL ansible tasks put inside `dev/ansible/` will be run when the box starts. We can use this to automatically start our module when we run `make up`. The ansible code for running terraform code is below. Add this to `run-terraform.yml` or another aptly named file.
```

Comment:
```suggestion
ALL ansible tasks inside `dev/ansible/` will be run when the box starts. We can use this to automatically start the module when we run `make up`. The ansible code for running the terraform code is below. Add this to `run-terraform.yml` or another file.
```

Reply:
This one has been rewritten. :) like the others; resolving